GUIDELINES.md

```diff
@@ -39,3 +39,70 @@ This file MUST NOT be exported publicly from instrumentation package, not direct
+
+Users and distributions need to install a version of `@opentelemetry/api` that is compatible with the instrumentation to use it.
+
+### OpenTelemetry Core packages
+
+Most instrumentations will depend on `@opentelemetry/instrumentation` and `@opentelemetry/semantic-conventions`. If needed, instrumentation can also depend on `@opentelemetry/core` for use of handy utils.
+
+Instrumentation SHOULD specify dependency as caret range (`^1.0.0`), with minimum supported version (and not latest version). If a specific feature or bug fix is required for instrumentation to function correctly, it SHOULD specify the minimum version that includes this feature.
+
+### Instrumented Package Dependency
+
+Instrumentations SHOULD NOT add a dependency on the package it is instrumenting as it can add large overhead for end users' application.
+
+This means that the instrumentation code SHOULD NOT `import` anywhere from the instrumented package. e.g. `@opentelemetry/instrumentation-foo` cannot `import 'foo'` as it might fail for applications that installed the instrumentation but not the `foo` package itself, which is a valid and supported use case for OpenTelemetry distributions and end users.
+
+It is allowed, however, to import `types`  from the instrumented package with the [`import type`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) syntax, as long as this type is not used in public api:
+
+```js
+// instrumentation.ts
+import type { Bar } from 'foo';
+```
+
+Since the instrumented package is installed as a dev dependency, types are available during development. Since they are not part of the public api, typescript removes these imports from the build artifacts during transpilation.
+
+To verify - `npm run compile` and check your `index.js` and `index.d.ts` files in the build directory. These files and any transitive files imported by them MUST NOT import the instrumented package.
+
+### Adding Types in Public API
+
+Sometimes, instrumented package types are needed in instrumentation's public api. These are mostly found in `types.ts` file on instrumentation config hooks that include data from the package and want to type it for consumers.
+
+To support this use case, you can choose one of the following options:
+
+1. Some packages do not distribute types. The types are alternatively available in the [Definitely Typed Project](https://github.com/DefinitelyTyped/DefinitelyTyped) as `@types/foo` (for a package named `foo`). Since @type package is mostly small in size, instrumentation MAY add dependency on the types package, and then use type-only import `import type from 'foo'` in `types.ts`.
+
+```js
+// package.json
+{
+    ...
+    "dependencies": {
+        "@types/foo": "1.2.3"
+    },
+    ...
+}
+
+// types.ts
+import type { Bar } from 'foo';
+
+export interface FooRequestInfo {
+    bar: Bar;
+}
+
+...
+```
```

Comment:
I believe that this block of code is not part of the list item above. If you were to indent this whole block by 4 it would be.

Reply:
Thanks, I learned something new :)